numpy/linalg/linalg.py

```diff
@@ -2512,18 +2510,12 @@ def _multi_dot_matrix_chain_order(arrays, return_costs=False):
 
     Also return the cost matrix if `return_costs` is `True`
 
-    The implementation CLOSELY follows Cormen, "Introduction to Algorithms",
-    Chapter 15.2, p. 370-378.  Note that Cormen uses 1-based indices.
-
-        cost[i, j] = min([
-            cost[prefix] + cost[suffix] + cost_mult(prefix, suffix)
-            for k in range(i, j)])
```

Comment:
Please don't remove this. The algorithm is presumably still based on Coram, but with some adaptations? The reference to source material is valuable.